src/msg/async/AsyncConnection.cc

```diff
@@ -2078,7 +2089,7 @@ void AsyncConnection::fault()
   }
 
   write_lock.Lock();
-  if (sd >= 0) {
+  if (sd >= 0 && state != STATE_CONNECTING_RE) {
```

Comment:
Oh. We need to reduce it. When we goto fail, we just convert to STATE_CONNECTING.